bin/publish-docs.sh

```diff
@@ -8,13 +8,13 @@ PACKAGE_VERSION=$(node -e 'console.log(require("./package").version)')
 
 git fetch origin gh-pages
 git checkout gh-pages
-git merge -
+git checkout main
+git branch -fd gh-pages
+git checkout -b gh-pages
 npm run public-docs
 sleep 1
-git rm -r docs
-sleep 1
 mv out docs
 sleep 1
 git add docs
 git commit -m "docs: update for ${PACKAGE_VERSION}"
-git push origin gh-pages
+git push --force origin gh-pages
```

Comment:
do we want to `--force-with-lease` instead?

Reply:
My understanding, based upon https://stackoverflow.com/a/52823955, is that it doesn't offer any benefit. We don't make any changes to the `gh-pages` branch other than to push up new API docs builds. Do you agree? Or do you still think we should try `--force-with-lease`?